docs/contributing/ui_guidelines.md

```diff
@@ -42,6 +43,36 @@ We use [Prettier](https://prettier.io/) for formatting and [ESLint](https://esli
 -   We follow a somewhat relaxed version of the [Airbnb styleguide](https://github.com/airbnb/javascript).
 -   Familiarise yourself with our [eslint-config-wagtail](https://github.com/wagtail/eslint-config-wagtail) configuration, which details our preferred code style.
 
+## Stimulus
+
+Wagtail uses [Stimulus](https://stimulus.hotwired.dev/) as a lightweight framework to attach interactive behaviour to DOM elements via `data-` attributes.
+
+### Why Stimulus
+
+Stimulus is a lightweight framework that allows developers to create interactive UI elements in a simple way. It makes it easy to do small-scale reactivity via changes to data attributes and does not require developers to 'init' things everywhere, unlike JQuery. It also provides an alternative to using inline script tag usage and window globals which reduces complexity in the codebase.
+
+### When to use Stimulus
+
+Stimulus is our [preferred library](https://github.com/wagtail/rfcs/pull/78) for simple client-side interactivity. It’s a good fit when:
+
+- The interactivity requires JavaScript. Otherwise, consider using HTML and CSS only.
+- Some of the logic is defined via HTML templates, not just JavaScript.
+- The interactivity is simple, and doesn’t require usage of more heavyweight libraries like React.
+
+Wagtail’s admin interface also leverages jQuery for similar scenarios. This is considered legacy and will eventually be removed. For new features, carefully consider whether existing jQuery code should be reused, or whether a rebuild with Stimulus is more appropriate.
+
+### How to build a Stimulus controller
+
+First think of how to name the controller. Keep it concise, one or two words ideally. Then,
+
+1. Start with the HTML templates, build as much of the UI as you can in HTML alone. Ensure it is accessible and follows the CSS guidelines.
+2. Create the controller file in our `client/src/controllers` folder, along with its tests (see [](testing)) and Storybook stories.
+3. For initialisation, consider which [controller lifecycle methods](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#methods) to use, if any (`connect`, `initialize`).
+4. If relevant, also consider how to handle the controlled element being removed from the DOM [`disconnect` lifecycle method](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#disconnection). 
+5. Document controller classes and methods with [JSDoc annotations](https://jsdoc.app/index.html).
+6. Arrange the `data attributes` in a consistent way. The `id`, `class`, and `type` comes first followed by `data-controller` and then alpha sorting of all other data attributes.
```

Comment:
```suggestion
6. Use [values](https://stimulus.hotwired.dev/reference/values) to provide options and also state. Only reach for instance properties for more complex internal state such as references to resolved DOM elements or other instances. Try to have falsey defaults and avoid too much usage of the Object type when using values.
```

I think we already cover the DOM attributes ordering above, but here is a replacement item 6 that may be useful.